forte/data/extractor/seqtagging_extractor.py

```diff
@@ -0,0 +1,203 @@
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+
+        Returns:
+            Feature: a feature that contains the extracted data.
+        """
+        instance_based_on: List[Annotation] = \
+            list(pack.get(self.config.based_on, instance))
+        instance_entry: List[Annotation] = \
+            list(pack.get(self.config.entry_type, instance))
+        instance_tagged = bio_tagging(pack, instance_based_on, instance_entry)
+
+        data = []
+        for pair in instance_tagged:
+            if pair[0] is None:
+                new_pair = (None, pair[1])
+            else:
+                new_pair = (getattr(pair[0], self.config.attribute), pair[1])
+            if self.vocab:
+                data.append(self.element2repr(new_pair))
+            else:
+                data.append(new_pair)
+        meta_data = {"pad_value": self.get_pad_value(),
+                     "dim": 1,
+                     "dtype": int if self.vocab else tuple}
+        return Feature(data=data,
+                       metadata=meta_data,
+                       vocab=self.vocab)
+
+    def pre_evaluation_action(self, pack: DataPack, instance: Annotation):
+        r"""This function is performed on the pack before the evaluation
+        stage, allowing one to perform some actions before the evaluation.
+        By default, this function will remove tags in the instance. You can
+        overwrite this function by yourself.
+
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance on which the
+                extractor performs the pre-evalation action.
+        """
+        for entry in pack.get(self.config.entry_type, instance):
+            pack.delete_entry(entry)
+
+    def add_to_pack(self, pack: DataPack, instance: Annotation,
+                    prediction: List[int]):
+        r"""Add the prediction for attribute to the instance.
+        We make following assumptions for prediction.
+        1. If we encounter "I" while its tag is different from the previous tag,
+           we will consider this "I" as a "B" and start a new tag here.
+        2. We will truncate the prediction it according to the number of entry.
+           If the prediction contains <PAD> element, this should remove them.
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance to which the
+                extractor add prediction.
+            prediction (Iterable[Union[int, Any]]): This is the output
+                of the model, which contains the index for attributes
+                of one instance.
+        """
+        instance_based_on = list(pack.get(self.config.based_on, instance))
```

Comment:
These variables will benefit from the explict typing.

Reply:
updated.